docs/fsharp/language-reference/span.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Span (F#)
+description: Learn how to create and use Span types in the F# programming language.
+ms.date: 07/23/2018
+---
+# Span<'T>
+
+## What is Span<'T>?
+
+`Span<'T>` is a value type that holds a pointer and a length value. It represents a slice/range of memory that a developer wants to work with in a type safe manner. The memory can be stack allocated, managed or unmanaged/native memory on the heap. It also has the same performance characterstics as an array.
+
+The type argument, `'T`, is the type of data we want to operate on from a `Span`.
+
+## Creating a Span<'T>
+
+A few typical ways of creating a `Span<'T>` are either by a managed array or a pointer.
+
+For example:
+```fsharp
```

Comment:
For larger examples, (well, really, any code that should be CI tested), we create the samples in the dotnet/samples repo (there is an fsharp directory, and the structure under it generally follows the structure of the docs). 

Then, we use the markdown extension to include the code into articles.

The advantage is that all the samples and snippets are build tested. And, we're slowly adding unit tests around the samples as well.

The one disadvantage is that it does mean two PRs in different repos, and it's harder to validate the includes until the samples are done.)